### ISSUE TITLE
Migrate Client().Update() calls back to Patch() calls

### ISSUE THREAD

**Issue by don-milos** [bluek8s/kubedirector#194]
When KubeDirector was migrated from operator-sdk version v0.0.6 to v0.8.1 the calls to sdk.Patch() were replaced with calls to Client().Update() because the Patch() API was removed. The Update() call replaces the entire resource object instead of a few selected fields in the object. Also, the objects now contain a ResourceVersion and if the update object is not at the latest ResourceVersion then the update fails with an IsConflict error with a message to try again with an up to date object. To recover from this error additional code was added to fetch the most recent version of the object, patch it, and then try the update again with the patched object.

The Patch() API is supposed to reintroduced in the next release of the K8s controller-runtime library. When a stable release with the Patch() API is available we need to port the KubeDirector back to using patch instead of update and remove all the retry logic from the code.

**Comment by joel-bluedata** [comment]
We can keep an eye on this file to see what version of controller-runtime is currently being used by the SDK: https://github.com/operator-framework/operator-sdk/blob/master/internal/pkg/scaffold/gopkgtoml.go

**Comment by joel-bluedata** [comment]
Next operator SDK release after v0.10.0 should using controller runtime 0.2.0.

**Comment by joel-bluedata** [comment]
Yep, releases v0.11.0 and later of the operator SDK use the controller-runtime that has PATCH support.

**Comment by joel-bluedata** [comment]
However when we update past v0.8.2, we'll have to finally transition from using dep to using Go modules instead, which, sigh. There's always something!

Created issue #222 for that.

**Comment by joel-bluedata** [comment]
Starting work on porting to new operator SDK in https://github.com/joel-bluedata/kubedirector/tree/sdk

**Comment by joel-bluedata** [comment]
Using new SDK now.

**Comment by joel-bluedata** [comment]
Not a blocker for 0.5.0, but it would be nice to tackle at least some of these so I will leave this on the 0.5.0 milestone for now.
